05-JWT-Basics/starter/controllers/main.js

```diff
@@ -0,0 +1,98 @@
+
+// if exist create new JWT
+// send back to frontend
+// setup authentication so only the request with JWT can access the dasboard
+
+
+
+const jwt = require('jsonwebtoken') // import the jwt package
+// const CustomAPIError = require('../errors/custom-error')
+const {BadRequestError} = require('../errors')
+
+//check in the controllers
+
+const login = async(req,res)=>{
+
+    const { username, password } = req.body //// check username, password in post(login) request
+    console.log(username,password)
+
+    //check if username and password exist - if not throw error
+    if (!username || !password) {
```

Comment:
In a production app, you'd likely build out this condition a bit to check for trailing strings, etc (so we don't create a user " "). Not required for this assignment though!

Reply:
Okay - thank you. Yes, I think I add more conditions in the preferred assignment but not here :)